File: src/main/java/com/terraViva/models/StudentScores.java
package com.terraViva.models;

import jakarta.persistence.*;
import lombok.*;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "student_scores")
public class StudentScores {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "student_id", nullable = false)
    private Student student;

    @ManyToOne
    @JoinColumn(name = "quiz_id", nullable = false)
    private Quiz quiz;

    private Integer score;
}

File: mobile/lib/models/StudentScores.dart

class StudentScores {
  final int studentId;
  final int quizId;
  final double score;

  StudentScores({
    required this.studentId,
    required this.quizId,
    required this.score,
  });

  factory StudentScores.fromJson(Map<String, dynamic> json) {
    return StudentScores(
      studentId: json['studentId'] as int,
      quizId: json['quizId'] as int,
      score: json['score'] as double,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'studentId': studentId,
      'quizId': quizId,
      'score': score,
    };
  }
}
